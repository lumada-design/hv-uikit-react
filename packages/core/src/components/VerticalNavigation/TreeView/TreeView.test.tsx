import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import {
  HvVerticalNavigationTreeView,
  HvVerticalNavigationTreeViewItem,
} from "@core/components";
import { Play, Stop } from "@hitachivantara/uikit-react-icons";

const Sample = () => (
  <HvVerticalNavigationTreeView selected="4" mode="navigation">
    <HvVerticalNavigationTreeViewItem icon={<Play />} nodeId="1" label="System">
      <HvVerticalNavigationTreeViewItem nodeId="2" label="SCPodF">
        <HvVerticalNavigationTreeViewItem nodeId="3" label="Compute" disabled />
        <HvVerticalNavigationTreeViewItem nodeId="4" label="Storage" />
        <HvVerticalNavigationTreeViewItem
          nodeId="5"
          label="Ethernet"
          selectable={false}
        />
        <HvVerticalNavigationTreeViewItem
          nodeId="6"
          label="Fiber Channel"
          payload={{ path: "/hello/world", params: { a: 2, b: "3" } }}
        />
      </HvVerticalNavigationTreeViewItem>
    </HvVerticalNavigationTreeViewItem>

    <HvVerticalNavigationTreeViewItem
      icon={<Stop />}
      nodeId="7"
      label="Administration"
    >
      <HvVerticalNavigationTreeViewItem nodeId="8" label="Rest API" />
      <HvVerticalNavigationTreeViewItem nodeId="9" label="License" />
      <HvVerticalNavigationTreeViewItem nodeId="10" label="Log Bundle" />
    </HvVerticalNavigationTreeViewItem>
  </HvVerticalNavigationTreeView>
);

describe("VerticalNavigation - Actions", () => {
  it("should be defined", () => {
    const { container } = render(<Sample />);
    expect(container).toBeDefined();
  });
});
