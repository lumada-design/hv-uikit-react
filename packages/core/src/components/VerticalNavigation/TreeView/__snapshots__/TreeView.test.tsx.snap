// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VerticalNavigation - Actions > should render correctly 1`] = `
<div>
  <ul
    class="HvVerticalNavigationTreeView-root css-1xsm1gc-StyledRoot e7gq1cg0"
  >
    <li
      class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
      data-hasicon="true"
      id="hvtreeview2-1"
    >
      <div
        class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
        id="hvtreeview2-1-button"
        role="button"
        style="padding-left: 10px;"
        tabindex="0"
      >
        <div
          class="css-n1g5h0-StyledIconsContainer emephz1"
        />
        <div
          class="css-1674i57-StyledLabel er1eqbm2"
        >
          System
        </div>
      </div>
      <ul
        class="HvVerticalNavigationTreeViewItem-group css-zinev9-StyledGroup er1eqbm3"
        id="hvtreeview2-1-group"
      >
        <li
          class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
          id="hvtreeview2-2"
        >
          <div
            class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
            id="hvtreeview2-2-button"
            role="button"
            style="padding-left: 20px;"
            tabindex="0"
          >
            <div
              class="css-n1g5h0-StyledIconsContainer emephz1"
            />
            <div
              class="css-1674i57-StyledLabel er1eqbm2"
            >
              SCPodF
            </div>
          </div>
          <ul
            class="HvVerticalNavigationTreeViewItem-group css-zinev9-StyledGroup er1eqbm3"
            id="hvtreeview2-2-group"
          >
            <li
              class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-disabled css-1y9s9wc-StyledNode er1eqbm1"
              id="hvtreeview2-3"
            >
              <div
                class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 HvTypography-disabled hv-1s9wi6o"
                id="hvtreeview2-3-button"
                role="button"
                style="padding-left: 30px;"
                tabindex="0"
              >
                <div
                  class="css-n1g5h0-StyledIconsContainer emephz1"
                />
                <div
                  class="css-1674i57-StyledLabel er1eqbm2"
                >
                  Compute
                </div>
              </div>
            </li>
            <li
              class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
              id="hvtreeview2-4"
            >
              <div
                class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
                id="hvtreeview2-4-button"
                role="button"
                style="padding-left: 30px;"
                tabindex="0"
              >
                <div
                  class="css-n1g5h0-StyledIconsContainer emephz1"
                />
                <div
                  class="css-1674i57-StyledLabel er1eqbm2"
                >
                  Storage
                </div>
              </div>
            </li>
            <li
              class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-unselectable css-1y9s9wc-StyledNode er1eqbm1"
              id="hvtreeview2-5"
            >
              <div
                class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
                id="hvtreeview2-5-button"
                role="button"
                style="padding-left: 30px;"
                tabindex="-1"
              >
                <div
                  class="css-n1g5h0-StyledIconsContainer emephz1"
                />
                <div
                  class="css-1674i57-StyledLabel er1eqbm2"
                >
                  Ethernet
                </div>
              </div>
            </li>
            <li
              class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
              id="hvtreeview2-6"
            >
              <div
                class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
                id="hvtreeview2-6-button"
                role="button"
                style="padding-left: 30px;"
                tabindex="0"
              >
                <div
                  class="css-n1g5h0-StyledIconsContainer emephz1"
                />
                <div
                  class="css-1674i57-StyledLabel er1eqbm2"
                >
                  Fiber Channel
                </div>
              </div>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li
      class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
      data-hasicon="true"
      id="hvtreeview2-7"
    >
      <div
        class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
        id="hvtreeview2-7-button"
        role="button"
        style="padding-left: 10px;"
        tabindex="0"
      >
        <div
          class="css-n1g5h0-StyledIconsContainer emephz1"
        />
        <div
          class="css-1674i57-StyledLabel er1eqbm2"
        >
          Administration
        </div>
      </div>
      <ul
        class="HvVerticalNavigationTreeViewItem-group css-zinev9-StyledGroup er1eqbm3"
        id="hvtreeview2-7-group"
      >
        <li
          class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
          id="hvtreeview2-8"
        >
          <div
            class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
            id="hvtreeview2-8-button"
            role="button"
            style="padding-left: 20px;"
            tabindex="0"
          >
            <div
              class="css-n1g5h0-StyledIconsContainer emephz1"
            />
            <div
              class="css-1674i57-StyledLabel er1eqbm2"
            >
              Rest API
            </div>
          </div>
        </li>
        <li
          class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
          id="hvtreeview2-9"
        >
          <div
            class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
            id="hvtreeview2-9-button"
            role="button"
            style="padding-left: 20px;"
            tabindex="0"
          >
            <div
              class="css-n1g5h0-StyledIconsContainer emephz1"
            />
            <div
              class="css-1674i57-StyledLabel er1eqbm2"
            >
              License
            </div>
          </div>
        </li>
        <li
          class="HvVerticalNavigationTreeViewItem-node HvVerticalNavigationTreeViewItem-selectable HvVerticalNavigationTreeViewItem-unselected css-1y9s9wc-StyledNode er1eqbm1"
          id="hvtreeview2-10"
        >
          <div
            class="HvTypography-root HvTypography-body HvVerticalNavigationTreeViewItem-content css-s2an9p-StyledContent er1eqbm0 hv-1gxwolr"
            id="hvtreeview2-10-button"
            role="button"
            style="padding-left: 20px;"
            tabindex="0"
          >
            <div
              class="css-n1g5h0-StyledIconsContainer emephz1"
            />
            <div
              class="css-1674i57-StyledLabel er1eqbm2"
            >
              Log Bundle
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
